### PR TITLE
Enable Blade on MacOS via "macos-blade" feature

### PR DESCRIPTION
**Pull request by kvark**
Depends on https://github.com/zed-industries/font-kit/pull/2 and https://github.com/kvark/blade/pull/77

This change enables Blade to be also used on MacOS. It will also make it easier to use it on Windows.

What works: most of the things. Zed loads as fast and appears equally responsive to the current renderer.
Screenshot 2024-02-11 at 12 09 15 AM

Things missing:
- [x] video streaming. ~~Requires a bit of plumbing on both Blade and Zed sides, but all fairly straightforward.~~
  -  verified with a local setup
- [x] resize. ~~Not sure where exactly to hook up the reaction on the window size change. Once we know where, the fix is one line.~~
- [ ] fine-tune CA Layer
  - this isn't a blocker for merging the PR, but it would be a blocker if we wanted to switch to the new path by default
- [ ] rebase on latest, get the dependency merged (need review/merge of https://github.com/zed-industries/font-kit/pull/2!)

Update: I implemented resize support as well as "surface" rendering on the Blade path (which will be useful on Linux/Windows later on). I haven't tested the latter though - not sure how to get something streaming. Would appreciate some help! I don't think this should be a blocker to this PR, anyway.

The only little piece that's missing for the Blade on MacOS path to be full-featured is fine-tuning the CALayer configuration. Zed does a lot of careful logic in configuring the layer, such as switching the "present with transaction" on/off intermittently, which Blade path doesn't have yet.

Release Notes:
- N/A

### CODE REVIEW

**Comment by mikayla-maki** [comment]
To test the livestreaming, we have scripts for running multiple Zed instances, but you need to have your enviroment setup a little bit. See [the instructions here](https://github.com/zed-industries/zed/blob/main/docs/src/developing_zed__local_collaboration.md) for more details :D.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -91,6 +91,9 @@ jobs:
             - name: Build other binaries
               run: cargo build --workspace --bins --all-features
 
+            - name: Build Blade version
```
Comment (mikayla-maki) [inline .github/workflows/ci.yml:94]: I don't think we'd want to add this to CI yet.

Comment (kvark) [reply .github/workflows/ci.yml:94]: Could we at least test `gpui` alone with this feature on CI (as opposed to full Zed)?

**Comment by kvark** [comment]
Was able to set up the local server and test the screen sharing with "macos-blade" - all looks good:
Screenshot 2024-02-12 at 11 33 15 PM

Gotta admit, it's very appealing as a tool for collaboration!

**Comment by kvark** [comment]
Dependencies are updated, code is rebased. Please take another look @mikayla-maki !

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mikayla-maki) [reply .github/workflows/ci.yml]: We've decided to keep it in CI, to make sure this is maintained :)

**Pull request merged**